src/main/java/org/assertj/core/presentation/CompositeRepresentation.java

```diff
@@ -0,0 +1,32 @@
+package org.assertj.core.presentation;
+
+import java.util.ArrayList;
+import java.util.Map;
+
+import static org.assertj.core.presentation.StandardRepresentation.STANDARD_REPRESENTATION;
+
+public class CompositeRepresentation implements Representation {
+  protected final Map<String, Representation> representationsByName;
+
+  public CompositeRepresentation(Map<String, Representation> representations) {
+    if (representations == null) throw new IllegalArgumentException();
+    this.representationsByName = representations;
+  }
+
+  @Override
+  public String toStringOf(Object object) {
```

Comment:
@joel-costigliola In this part I thought priority could be determined by the insertion order directly, so I did not add a `priority` field in `Representation`.  So we actually need to add a field called `priority` in `Representatioin`?  And I do not fully understand when  should we print a message, in that case, saying that there were 2 representations with the same priority? Could you please explain more details? Thank you very much! 😊

Reply:
@joel-costigliola Hi! I am very sorry to bother you again because I am not sure whether you have seen this question since you answered another question related to this pr several hours before but this one remains silent. Looking forward to your reply! 😉

Reply:
Since the `Representation` is an `Interface`, where I believe a `int priority` field can not be added to implement this mechanism, I wonder if it is I got a wrong idea or we have to think about another implementation of the `priority` part.☺️😳